### PR TITLE
Fixed circular symlink infinte loop

### PR DESCRIPTION
**Pull request by airarrazaval**
Fix for issue https://github.com/mihneadb/node-directory-tree/issues/89

Other improvements in this PR:

* Fixed indentation in `index.d.ts`
* Improved types definition
* Added ` isSymbolicLink`  property for DirectoryTree item and `followSymlinks` property in DirectoryTree options

### CODE REVIEW

File: lib/directory-tree.js [hunk 1]
```diff
@@ -50,34 +50,55 @@ function isRegExp(regExp) {
  */
 function directoryTree (path, options, onEachFile, onEachDirectory) {
 	const name = PATH.basename(path);
-	path = options && options.normalizePath ? normalizePath(path) : path;
+  options = options || {};
```
Comment (mihneadb) [inline lib/directory-tree.js:53]: The indentation looks odd here.

**Review by mihneadb — CHANGES_REQUESTED** [review]
Thanks for this! 2 minor asks:
* please make sure to fix the indentation in the js file
* a test would be nice (can't remember if git supports symlinks or not?)

**Review by airarrazaval — COMMENTED** [review]
I fixed the indentation by replacing tabs with spaces. Regarding symlink, git does support it.  I will have a look at it.

**Comment by airarrazaval** [comment]
@mihneadb Tests successfully passing and included symlink

**Review by mihneadb — APPROVED** [review]
Great, thanks!

File: index.d.ts [hunk 2]
```diff
@@ -1,58 +1,32 @@
...
+    size: number;
+    type: "directory" | "file";
+    children ? : DirectoryTree[];
+    extension?: string;
+    isSymbolicLink?: boolean;
+  }
+  export interface DirectoryTreeOptions {
+    normalizePath ? : boolean;
+    exclude ? : RegExp | RegExp[];
+    attributes ? : (keyof directoryTree.Stats)[];
```
Comment (mihneadb) [inline index.d.ts:25]: directoryTree.Stats is no longer a thing I guess. 😅 https://github.com/mihneadb/node-directory-tree/issues/91

**Pull request merged**